arkouda/pdarraycreation.py

```diff
@@ -821,3 +820,137 @@ def random_strings_lognormal(logmean : numeric_scalars, logstd : numeric_scalars
                  NUMBER_FORMAT_STRINGS['float64'].format(logstd),
                  seed))
     return Strings(*(cast(str,repMsg).split('+')))
+
+
+
+@typechecked
+def suffix_array(strings : Strings) -> SArrays:
+        """
+        Return the suffix arrays of given strings. The size/shape of each suffix
+	arrays is the same as the corresponding strings. 
+	A simple example of suffix array is as follow. Given a string "banana$",
+	all the suffixes are as follows. 
+	s[0]="banana$"
+	s[1]="anana$"
+	s[2]="nana$"
+	s[3]="ana$"
+	s[4]="na$"
+	s[5]="a$"
+	s[6]="$"
+	The suffix array of string "banana$"  is the array of indices of sorted suffixes.
+	s[6]="$"
+	s[5]="a$"
+	s[3]="ana$"
+	s[1]="anana$"
+	s[0]="banana$"
+	s[4]="na$"
+	s[2]="nana$"
+	so sa=[6,5,3,1,0,4,2]
+
+        Returns
+        -------
+        pdarray
+            The suffix arrays of the given strings
+
+        See Also
+        --------
+
+        Notes
+        -----
+        
+        Raises
+        ------  
+        RuntimeError
+            Raised if there is a server-side error in executing group request or
+            creating the pdarray encapsulating the return message
+        """
+        cmd=  "segmentedSuffixAry"
+        args = "{} {} {}".format( strings.objtype,
+                                                        strings.offsets.name,
+                                                        strings.bytes.name) 
+        repMsg = generic_msg(cmd=cmd,args=args)
+        return SArrays(*(cast(str,repMsg).split('+')))
```

Comment:
While not the biggest deal, there are a handful of place where spacing/formatting differs from the rest of the arkouda codebase. This is a bit of nitpick from me but I think having consistency is important for readability/maintainability

I would likely update this to look something like this
```python
cmd = "segmentedSuffixAry"
args = "{} {} {}".format(strings.objtype, strings.offsets.name, strings.bytes.name)
repMsg = generic_msg(cmd=cmd, args=args)
return SArrays(*(cast(str, repMsg).split("+")))

```